### PR TITLE
Synthetic package.json cleanup + Yarn upgrade

### PR DESCRIPTION
**Pull request by mnonnenmacher**
See the commit messages for details.

### CODE REVIEW

File: analyzer/src/funTest/assets/projects/synthetic/npm-version-urls/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1524 @@
+{
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/npm-version-urls/package-lock.json]: Is the addition of this file intended? At least it does not seem to match the commit's purpose as outlined in the commit message.

File: analyzer/src/funTest/assets/projects/synthetic/npm/no-lockfile/package.json [hunk 2]
```diff
@@ -1,43 +1,11 @@
 {
   "name": "npm-no-lockfile",
-  "version": "2.0.0",
+  "version": "1.0.0",
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/synthetic/npm/no-lockfile/package.json]: Nit: Again a change instead of removal.

Comment (sschuberth) [reply analyzer/src/funTest/assets/projects/synthetic/npm/no-lockfile/package.json]: Also several times more below.

File: .appveyor.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ environment:
   SBT_VERSION: 1.0.2
   STACK_VERSION: 2.1.3.20190715
   VIRTUAL_ENV_VERSION: 15.1.0
-  YARN_VERSION: 1.17.3
+  YARN_VERSION: 1.21.1
```
Comment (sschuberth) [inline .appveyor.yml]: Just wondering, is there a concrete reason for the upgrade? Like older Yarn versions not being available anymore in some distros?

File: analyzer/src/funTest/assets/projects/synthetic/npm-babel/package.json [hunk 4]
```diff
@@ -5,7 +5,6 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1"
   },
```
[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/npm-babel/package.json]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/npm-expected-output.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ---
 project:
-  id: "NPM::npm-project:2.0.0"
+  id: "NPM::npm-project:1.0.0"
```
[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/npm-expected-output.yml]: Done.

File: analyzer/src/funTest/assets/projects/synthetic/npm-version-urls/package-lock.json (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/assets/projects/synthetic/npm-version-urls/package-lock.json]: No, I have removed it.

File: .appveyor.yml (hunk 3 above)
Comment (mnonnenmacher) [reply .appveyor.yml]: No, it's just to keep our dependencies up to date.

[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply .appveyor.yml]: Done.

**Pull request merged**